akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -564,10 +570,10 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
       log.debug("ShardRegion terminated: [{}]", ref)
       state.regions(ref).foreach { s ⇒ self ! GetShardHome(s) }
 
-      gracefulShutdownInProgress -= ref
```

Comment:
There is no difference in practise, but it feels (reads) better to modify the state of the actor after persist of ShardRegionTerminated has been done. The reason it doesn't matter is that no other message is processed in-between (stashing) and if the persist fails the actor will be stopped anyway.